pkg/cmd/kind/build/nodeimage/nodeimage.go

```diff
@@ -97,7 +94,7 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 	if err := nodeimage.Build(
 		nodeimage.WithImage(flags.Image),
 		nodeimage.WithBaseImage(flags.BaseImage),
-		nodeimage.WithKuberoot(kubeRoot),
+		nodeimage.WithKubeParam(kubeRoot),
```

Comment:
@BenTheElder it seems with this approach we infer from the input if we should: build from source, fetch from url or extract from tarball

Is it too magical? should we recover type ?
At least we should document it , the `Use: ` ....

better wait for @BenTheElder he is good with these cli UX things

Reply:
definitely! no rush.

Reply:
> it seems with this approach we infer from the input if we should: build from source, fetch from url or extract from tarball

I think we should infer it and allow the user to override the inference.

We can say:
- empty input == source build and autodetect
- starts with a drive letter, `/`, or `.` => source build from the path
- build from a release for a parseable kube version OR a few magical well-known like the "latest" tags

- `--type` explicitly overrides this argument parsing

how about that? Then the common path is just like `kind build node-image .`, `kind build node-image $HOME/kubernetes`, `kind build node-image v1.28.2` and only the confusing case of "I have a directory named v1.28.2 and want to do ./v1.28.2 but pass it as `v1.28.2` requires setting `--type` or users that want to be certain and avoid any chance.

Reply:
We could have similar detection for "is a release tar" if we want to support that.

Reply:
consolidated table https://github.com/kubernetes-sigs/kind/pull/3614#issuecomment-2121474989